clients/roscpp/src/libros/init.cpp

```diff
@@ -612,4 +612,9 @@ void shutdown()
   Time::shutdown();
 }
 
+const std::string& getDefaultMasterURI() {
+  static const std::string uri = "http://localhost:11311";
```

Comment:
In the spirit of not spreading around the constant (as that was raised in the original review as a concern), what if we used `execute_process` from CMake to generate the implementation for this function based on a Python script, which could then import the `rosgraph.rosenv.DEFAULT_MASTER_URI` value?

Reply:
Although this would be a feasible way, I'd argue, that this would a require a solution, which is too complex for this particular case. There might be other constants, which are currently defined in roscpp and rospy (or anywhere else but in two different places). I think, it would be beneficial to find a solution for all these cases.

I'd suggest to merge this PR as it is for now and then think about a general solution for unifying the constants of C++ and Python components.

Reply:
I don't think it would really be that complicated, but I also don't feel strongly about it. :)